shared/modules/consensus_module.go

```diff
@@ -91,10 +88,11 @@ type ConsensusDebugModule interface {
 
 	SetUtilityUnitOfWork(UtilityUnitOfWork)
 
-	// SetAggregatedStateSyncMetadata is used to set peer's aggregated metadata in testing scenarios to simulate periodic metadata synchronization. It is not intended to be used outside of testing.
-	SetAggregatedStateSyncMetadata(minHeight, maxHeight uint64, peerAddress string)
-	GetAggregatedStateSyncMetadataMaxHeight() (minHeight uint64)
+	// PushStateSyncMetadataResponse is used to simulate metadata received metadata in testing scenarios. It is not intended to be used outside of testing.
+	PushStateSyncMetadataResponse(minHeight, maxHeight uint64)
```

Comment:
We have a structure for metadata responses (i.e. the protobuf).

You should be pushing those instead.

Reply:
Agreed, please see my answer below.